web/html/src/manager/content-management/list-filters/filter.utils.test.js

```diff
@@ -0,0 +1,169 @@
+  });
+
+  test('test filter advisory name', () => {
+    const filterForm = {
+      filter_name: "filter advisory_name name",
+      matcher: "equals",
+      type: "advisory_name",
+      advisory_name: "advisory_name_12",
+      rule: "deny"
+    }
+    expect(mapResponseToFilterForm([mapFilterFormToRequest(filterForm)])[0])
+      .toEqual(expect.objectContaining(filterForm));
+    expect(mapFilterFormToRequest(filterForm).criteriaKey).toEqual("advisory_name");
+    expect(mapFilterFormToRequest(filterForm).criteriaValue).toEqual("advisory_name_12");
+  });
+
+  test('test filter advisory type', () => {
+    const filterForm = {
+      filter_name: "filter advisory_type name",
+      matcher: "equals",
+      type: "advisory_type",
+      advisory_type: "Security Advisory",
+      rule: "deny"
+    }
+    expect(mapResponseToFilterForm([mapFilterFormToRequest(filterForm)])[0])
+      .toEqual(expect.objectContaining(filterForm));
+    expect(mapFilterFormToRequest(filterForm).criteriaKey).toEqual("advisory_type");
+    expect(mapFilterFormToRequest(filterForm).criteriaValue).toEqual("Security Advisory");
+
+    filterForm.rule = "allow";
+    filterForm.advisory_type = "Product Enhancement Advisory";
+
+    expect(mapResponseToFilterForm([mapFilterFormToRequest(filterForm)])[0])
+      .toEqual(expect.objectContaining(filterForm));
+    expect(mapFilterFormToRequest(filterForm).criteriaKey).toEqual("advisory_type");
+    expect(mapFilterFormToRequest(filterForm).criteriaValue).toEqual("Product Enhancement Advisory");
+  });
+
+  test('test filter synopsis', () => {
+    const filterForm = {
+      filter_name: "filter synopsis name",
+      matcher: "equals",
+      type: "synopsis",
+      synopsis: "synopsis_name",
+      rule: "deny"
+    }
+    expect(mapResponseToFilterForm([mapFilterFormToRequest(filterForm)])[0])
+      .toEqual(expect.objectContaining(filterForm));
+    expect(mapFilterFormToRequest(filterForm).criteriaKey).toEqual("synopsis");
+    expect(mapFilterFormToRequest(filterForm).criteriaValue).toEqual("synopsis_name");
+    expect(mapFilterFormToRequest(filterForm).matcher).toEqual("equals");
+
+    filterForm.rule = "allow";
+    filterForm.matcher = "contains";
+
+    expect(mapResponseToFilterForm([mapFilterFormToRequest(filterForm)])[0])
+      .toEqual(expect.objectContaining(filterForm));
+    expect(mapFilterFormToRequest(filterForm).criteriaKey).toEqual("synopsis");
+    expect(mapFilterFormToRequest(filterForm).criteriaValue).toEqual("synopsis_name");
+    expect(mapFilterFormToRequest(filterForm).matcher).toEqual("contains");
+    expect(mapFilterFormToRequest(filterForm).rule).toEqual("allow");
+  });
+
+  test('test patch by date', () => {
+    const filterForm = {
+      filter_name: "filter by date",
+      matcher: "equals",
+      type: "issue_date",
+      issue_date: "2017-10-01T00:00+01",
+      rule: "deny"
+    }
+    // TODO: add this test when moment isn't a global dependency and comes from NPM
```

Comment:
I think that we should:
- either track this task in a github tech-debt issue,
- or delete this test.

Otherwise I'm pretty sure we'll forget to address the `TODO` :smile: